File: mobile/lib/shared/models/milk_entry.dart
class MilkEntry {
  final String id;
  final String customerId;
  final DateTime deliveryDate;
  final double quantityLitres;
  final double pricePerLitre;
  final double totalAmount;

  MilkEntry({
    required this.id,
    required this.customerId,
    required this.deliveryDate,
    required this.quantityLitres,
    required this.pricePerLitre,
    required this.totalAmount,
  });

  factory MilkEntry.fromJson(Map<String, dynamic> json) => MilkEntry(
        id: json['id'],
        customerId: json['customer_id'],
        deliveryDate: DateTime.parse(json['delivery_date']),
        quantityLitres: (json['quantity_litres'] as num).toDouble(),
        pricePerLitre: (json['price_per_litre'] as num).toDouble(),
        totalAmount: (json['total_amount'] as num).toDouble(),
      );
}

File: backend/app/schemas/milk_entry.py
from datetime import date, datetime
from pydantic import BaseModel


class MilkEntryCreate(BaseModel):
    customer_id: str
    delivery_date: date
    quantity_litres: float


class MilkEntryUpdate(BaseModel):
    quantity_litres: float | None = None
    delivery_date: date | None = None


class MilkEntryOut(BaseModel):
    id: str
    customer_id: str
    milkman_id: str
    delivery_date: date
    quantity_litres: float
    price_per_litre: float
    total_amount: float
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
